haskell.el

```diff
@@ -312,7 +312,7 @@
   (let ((ident (haskell-ident-at-point))
         (tags-file-name (haskell-session-tags-filename (haskell-session)))
         (tags-revert-without-query t))
-    (when (not (string= "" (haskell-string-trim ident)))
+    (when (and (stringp ident) (not (string= "" (haskell-string-trim ident))))
```

Comment:
Change:

```el
(when (and (stringp ident) (not (string= "" (haskell-string-trim ident)))
```
into

```el
(when (and ident (not (string= "" (haskell-string-trim ident)))
```

`ident` should be either `nil` or `string` at this point. I would prefer it to crash if it is something else rather than silently swallow wrong type.